lib/auth/auth_with_roles.go

```diff
@@ -6238,7 +6248,17 @@ func (a *ServerWithRoles) CreatePrivilegeToken(ctx context.Context, req *proto.C
 
 // CreateRegisterChallenge is implemented by AuthService.CreateRegisterChallenge.
 func (a *ServerWithRoles) CreateRegisterChallenge(ctx context.Context, req *proto.CreateRegisterChallengeRequest) (*proto.MFARegisterChallenge, error) {
-	// The token provides its own authorization and authentication.
+	switch {
+	case req.TokenID != "":
+	case req.ExistingMFAResponse != nil:
+		if a.hasBuiltinRole(types.RoleProxy) {
```

Comment:
You mean authz.IsLocalUser, right? I'm afraid it could have knock-on effects on... unusual trusted cluster setups, but I suppose I could check for both local and remote. Let me give that a try.

Reply:
Done, PTAL.

Reply:
Actually this broke a bunch of a tests, let me take a look first.

Reply:
I had the ifs inverted. 🤦 

PTAL.